gravitee-apim-console-webui/src/management/settings/top-apis/migrated/top-apis.component.ts

```diff
@@ -0,0 +1,152 @@
+import { MatDialog } from '@angular/material/dialog';
+import { EMPTY, Observable, Subject } from 'rxjs';
+import { catchError, filter, switchMap, takeUntil, tap } from 'rxjs/operators';
+import { sortBy } from 'lodash';
+import { GIO_DIALOG_WIDTH, GioConfirmDialogComponent, GioConfirmDialogData } from '@gravitee/ui-particles-angular';
+
+import { TopApi } from './top-apis.model';
+import { TopApisDialogComponent } from './top-apis-dialog/top-apis-dialog.component';
+
+import { SnackBarService } from '../../../../services-ngx/snack-bar.service';
+import { TopApiService } from '../../../../services-ngx/top-api.service';
+import { Api } from '../../../../entities/api';
+
+@Component({
+  selector: 'app-top-apis',
+  templateUrl: './top-apis.component.html',
+  styleUrls: ['./top-apis.component.scss'],
+})
+export class TopApisComponent implements OnInit, OnDestroy {
+  private unsubscribe$: Subject<boolean> = new Subject<boolean>();
+  public topApisList: TopApi[] = [];
+  public displayedColumns: string[] = ['picture', 'name', 'version', 'description', 'actions'];
+
+  constructor(
+    public topApiService: TopApiService,
+    private snackBarService: SnackBarService,
+    private matDialog: MatDialog,
+  ) {}
+
+  ngOnInit(): void {
+    this.topApiService.getList().subscribe((topApisList: TopApi[]): void => {
+      this.topApisList = topApisList;
+    });
+  }
+
+  ngOnDestroy(): void {
+    this.unsubscribe$.next(true);
+    this.unsubscribe$.unsubscribe();
+  }
+
+  public isFirst(order: number): boolean {
+    return order === 0;
+  }
+
+  public isLast(order: number): boolean {
+    return order === this.topApisList.length - 1;
+  }
+
+  public moveUp(topApi: TopApi): void {
+    if (topApi.order > 0) {
+      this.changeOrder(topApi.order, topApi.order - 1);
+    }
+  }
+
+  public moveDown(topApi: TopApi): void {
+    if (topApi.order < this.topApisList.length - 1) {
+      this.changeOrder(topApi.order, topApi.order + 1);
+    }
+  }
+
+  private changeOrder(oldOrder: number, newOrder: number): void {
+    this.topApisList[oldOrder].order = newOrder;
+    this.topApisList[newOrder].order = oldOrder;
+    this.topApisList = sortBy(this.topApisList, 'order');
+
+    this.updateTopApisList().subscribe((list: TopApi[]): void => {
+      this.topApisList = list;
+    });
+  }
+
+  private updateTopApisList(): Observable<TopApi[]> {
+    return this.topApiService.update(this.topApisList).pipe(
+      tap((): void => {
+        this.snackBarService.success('List updated successfully');
+      }),
+      catchError(({ error }) => {
+        this.snackBarService.error(error.message);
+        return EMPTY;
+      }),
+      takeUntil(this.unsubscribe$),
+    );
+  }
+
+  public deleteTopApi(element: TopApi): void {
+    this.matDialog
+      .open<GioConfirmDialogComponent, GioConfirmDialogData, boolean>(GioConfirmDialogComponent, {
+        data: {
+          title: 'Delete Top API',
```

Comment:
hum maybe more ttl : `Delete from the list of Top APIs` WDYT ? 
because we're not deleting the API here. it's confusing

Reply:
I was going to recommend `Remove from Top APIs`. 😄

Reply:
Make sense to rename `deletion` to `removing` from list